### PR TITLE
Metadata partition

### PR DESCRIPTION
**Pull request by AmandeepSinghCS**

### CODE REVIEW

File: clouddq/classes/dataplex_entity_schema.py [hunk 1]
```diff
@@ -38,21 +42,43 @@ def from_dict(
             error_msg=f"Entity {entity_id}: must define non-empty value: 'fields'.",
         )
 
+        if kwargs.get("partitionFields"):
+            partition_fields = kwargs.get("partitionFields")
```
Comment (thinhha) [inline clouddq/classes/dataplex_entity_schema.py:46]: where are you parsing this into DataplexEntityPartitionSchemaField?

File: clouddq/classes/dataplex_entity_schema.py [hunk 2]
```diff
@@ -38,21 +42,43 @@ def from_dict(
...
-          self: DataplexEntitySchemaField:
+          self: DataplexEntitySchema:
 
         Returns:
 
         """
 
         output = {
             "fields": self.fields,
+            "partitionFields": self.partitionFields,
```
Comment (thinhha) [inline clouddq/classes/dataplex_entity_schema.py:80]: do you need to call DataplexEntityPartitionSchemaField.to_dict() here?

File: clouddq/classes/dataplex_entity_schema_partition_fields.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        return dict(output)
+
+    @classmethod
+    def from_dict(
+        cls: DataplexEntityPartitionSchemaField, kwargs: dict
+    ) -> DataplexEntityPartitionSchemaField:
+
+        name = kwargs.get("name")
+        assert_not_none_or_empty(
+            value=name, error_msg="Name: must define non-empty value: 'name'."
```
Comment (thinhha) [inline clouddq/classes/dataplex_entity_schema_partition_fields.py:53]: Please improve error message here to say something like:
"DataplexEntity {entity_id}: must define non-empty value: ..."

File: clouddq/classes/dataplex_entity_schema.py [hunk 4]
```diff
@@ -38,21 +42,43 @@ def from_dict(
...
+            partition_style = kwargs.get("partitionStyle", None)
+            assert_not_none_or_empty(
+                value=partition_style,
+                error_msg=f"Entity {entity_id}: must define non-empty value: 'partition_style'."
+            )
+        else:
+            partition_style = None
+
         return DataplexEntitySchema(
             fields=fields,
```
Comment (thinhha) [inline clouddq/classes/dataplex_entity_schema.py:64]: ditto here do you need to parse fields into DataplexEntitySchemaField?

File: clouddq/classes/dataplex_entity_schema.py [hunk 5]
```diff
@@ -38,21 +42,43 @@ def from_dict(
...
         Args:
-          self: DataplexEntitySchemaField:
+          self: DataplexEntitySchema:
 
         Returns:
 
         """
 
         output = {
             "fields": self.fields,
```
Comment (thinhha) [inline clouddq/classes/dataplex_entity_schema.py:79]: ditto here do you need to call DataplexEntitySchemaField.to_dict() here?

File: tests/resources/dataplex_gcs_metadata_expected_configs.json [hunk 6]
```diff
@@ -27,6 +27,7 @@
     "dataset_name": "<your_dataplex_bigquery_dataset_id>",
     "project_name": "<your-gcp-project-id>",
     "resource_type": "CLOUD_STORAGE",
+    "partition_fields": null,
```
Comment (thinhha) [inline tests/resources/dataplex_gcs_metadata_expected_configs.json:30]: can we change this GCS asset to one that has a partition field? I'd like to have a test for this.

**Review by thinhha — COMMENTED** [review]
thanks Amandeep. I have a few comments/feedback. Could you please take a look?

File: tests/resources/dataplex_gcs_metadata_expected_configs.json (hunk 6 above)
Comment (AmandeepSinghCS) [reply tests/resources/dataplex_gcs_metadata_expected_configs.json:30]: I am adding a separate test for the partitioned gcs asset, this is for the existing one without partitioning.

File: clouddq/classes/dq_configs_cache.py [hunk 7]
```diff
@@ -67,6 +68,10 @@ def get_table_entity_id(self, entity_id: str) -> dq_entity.DqEntity:
         convert_json_value_to_dict(entity_record, "environment_override")
         convert_json_value_to_dict(entity_record, "columns")
         entity = dq_entity.DqEntity.from_dict(entity_id, entity_record)
+        partition_fields = entity.partition_fields
+        if partition_fields:
+            partition_fields = json.loads(partition_fields)
+            entity.partition_fields = partition_fields
```
[the remark this answers is not in the document]
Comment (AmandeepSinghCS) [reply clouddq/classes/dq_configs_cache.py:74]: sure.

File: clouddq/classes/dataplex_entity_schema_partition_fields.py (hunk 3 above)
Comment (AmandeepSinghCS) [reply clouddq/classes/dataplex_entity_schema_partition_fields.py:53]: done

File: clouddq/classes/dataplex_entity_schema.py (hunk 1 above)
Comment (AmandeepSinghCS) [reply clouddq/classes/dataplex_entity_schema.py:46]: parsed it to DataplexEntityPartitionSchemaField class

File: clouddq/classes/dataplex_entity_schema.py (hunk 2 above)
Comment (AmandeepSinghCS) [reply clouddq/classes/dataplex_entity_schema.py:80]: calling the to_dict while parsing

File: clouddq/classes/dataplex_entity_schema.py (hunk 4 above)
Comment (AmandeepSinghCS) [reply clouddq/classes/dataplex_entity_schema.py:64]: parsed to DataplexEntitySchemaField class

File: clouddq/classes/dataplex_entity_schema.py (hunk 5 above)
Comment (AmandeepSinghCS) [reply clouddq/classes/dataplex_entity_schema.py:79]: calling it while parsing the fields.

**Review by thinhha — APPROVED** [review]
LGTM - thanks!

**Pull request merged**